securedrop_export/main.py

```diff
@@ -1,46 +1,180 @@
-        action = PrintTestPageAction(submission)
-
-    action.run()
+        safe_mkdir(DEFAULT_HOME)
+        safe_mkdir(DEFAULT_HOME, LOG_DIR_NAME)
+
+        log_file = os.path.join(DEFAULT_HOME, LOG_DIR_NAME, EXPORT_LOG_FILENAME)
+
+        # set logging format
+        log_fmt = (
+            "%(asctime)s - %(name)s:%(lineno)d(%(funcName)s) "
+            "%(levelname)s: %(message)s"
+        )
+        formatter = logging.Formatter(log_fmt)
+
+        handler = TimedRotatingFileHandler(log_file)
+        handler.setFormatter(formatter)
+
+        # For rsyslog handler
+        if platform.system() != "Linux":  # pragma: no cover
+            syslog_file = "/var/run/syslog"
+        else:
+            syslog_file = "/dev/log"
+
+        sysloghandler = SysLogHandler(address=syslog_file)
+        sysloghandler.setFormatter(formatter)
+        handler.setLevel(logging.DEBUG)
+
+        # set up primary log
+        log = logging.getLogger()
+        log.setLevel(logging.DEBUG)
+        log.addHandler(handler)
+        # add the second logger
+        log.addHandler(sysloghandler)
+    except Exception as ex:
+        raise ExportException(sdstatus=Status.ERROR_LOGGING) from ex
+
+
+def _start_service(submission: Archive) -> Status:
+    """
+    Start print or export service.
+    """
+    # Print Routines
+    if submission.command is Command.PRINT:
+        return PrintService(submission).print()
+    elif submission.command is Command.PRINTER_PREFLIGHT:
+        return PrintService(submission).printer_preflight()
+    elif submission.command is Command.PRINTER_TEST:
+        return PrintService(submission).printer_test()
+
+    # Export routines
+    elif submission.command is Command.EXPORT:
+        return ExportService(submission).export()
+    elif submission.command is Command.CHECK_USBS:
+        return ExportService(submission).check_connected_devices()
+    elif submission.command is Command.CHECK_VOLUME:
+        return ExportService(submission).check_disk_format()
+
+
+def _exit_gracefully(submission: Archive, status: BaseStatus = None):
+    """
+    Write status code, ensure file cleanup, and exit with return code 0.
+    Non-zero exit values will cause the system to try alternative
+    solutions for mimetype handling, which we want to avoid.
+    """
+    if status:
+        logger.info(f"Exit gracefully with status: {status.value}")
+    else:
+        logger.info("Exit gracefully (no status code supplied)")
+    try:
+        # If the file archive was extracted, delete before returning
+        if submission and os.path.isdir(submission.tmpdir):
+            shutil.rmtree(submission.tmpdir)
+        # Do this after deletion to avoid giving the client two error messages in case of the
+        # block above failing
+        _write_status(status)
+    except Exception as ex:
+        logger.error("Unhandled exception: {}".format(ex))
+        logger.error(ex)
```

Comment:
Why two log entries?

Reply:
mistake! addressed in [d0fcf68](https://github.com/freedomofpress/securedrop-export/pull/105/commits/d0fcf68d9c856ef3e622a806829ec7e86bf6d497)